cypress/integration/page.spec.js

```diff
@@ -1,7 +1,290 @@
+      page.sortByLengthButton().should('have.class', NOT_ACTIVE_CLASS);
+    });
+
+    it('should have active "Reverse" button', () => {
+      page.reverseButton().should('not.have.class', NOT_ACTIVE_CLASS);
+    });
+
+    it('should allow to sort alphabetically', () => {
+      page.sortAlphButton().click();
+
+      page.assertFirstGoods('Jam', 'Ice cream', 'Honey');
+      page.reverseButton().should('not.have.class', NOT_ACTIVE_CLASS);
+      page.sortAlphButton().should('not.have.class', NOT_ACTIVE_CLASS);
+    });
+
+    it('should allow to sort by length', () => {
+      page.sortByLengthButton().click();
+
+      page.assertFirstGoods('Ice cream', 'Dumplings', 'Garlic');
+      page.reverseButton().should('not.have.class', NOT_ACTIVE_CLASS);
+      page.sortByLengthButton().should('not.have.class', NOT_ACTIVE_CLASS);
+    });
+
+    it('should allow to reset', () => {
+      page.resetButton().click()
+
+      page.assertFirstGoods('Dumplings', 'Carrot', 'Eggs');
+      page.sortAlphButton().should('have.class', NOT_ACTIVE_CLASS);
+      page.sortByLengthButton().should('have.class', NOT_ACTIVE_CLASS);
+      page.reverseButton().should('have.class', NOT_ACTIVE_CLASS);
+      page.resetButton().should('not.exist');
+    });
+  });
+
+  describe('after double reverse', () => {
+    beforeEach(() => {
+      page.reverseButton().click();
+      page.reverseButton().click();
+    });
+
+    it(`should show goods in the original order`, () => {
+      page.assertFirstGoods('Dumplings', 'Carrot', 'Eggs');
+    });
+
+    it('should not show "Reset" button', () => {
+      page.resetButton().should('not.exist');
+    });
+
+    it('should have not active "Sort alphabetically" button', () => {
+      page.sortAlphButton().should('have.class', NOT_ACTIVE_CLASS);
+    });
+
+    it('should have not active "Sort by length" button', () => {
+      page.sortByLengthButton().should('have.class', NOT_ACTIVE_CLASS);
+    });
+
+    it('should have not active "Reverse" button', () => {
+      page.reverseButton().should('have.class', NOT_ACTIVE_CLASS);
+    });
+  });
+
+  describe('after pressing "Sort alphabetically"', () => {
+    beforeEach(() => {
+      page.sortAlphButton().click();
+    });
+
+    it(`should sort goods alphabetically`, () => {
+      page.assertFirstGoods('Apple', 'Bread', 'Carrot');
+    });
+
+    it('should show "Reset" button', () => {
+      page.resetButton().should('exist');
+    });
+
+    it('should have active "Sort alphabetically" button', () => {
+      page.sortAlphButton().should('not.have.class', NOT_ACTIVE_CLASS);
+    });
+
+    it('should have not active "Sort by length" button', () => {
+      page.sortByLengthButton().should('have.class', NOT_ACTIVE_CLASS);
+    });
+
+    it('should have not active "Reverse" button', () => {
+      page.reverseButton().should('have.class', NOT_ACTIVE_CLASS);
+    });
+
+    it('should allow to reset', () => {
+      page.resetButton().click()
```

Comment:
Missing semicolon at the end of the line. Consider adding it to maintain consistency and avoid potential issues.